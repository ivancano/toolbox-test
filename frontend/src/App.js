import React from 'react';
import { Row, Col, Container, Form, Button } from 'react-bootstrap';
import {inject, observer, Provider} from "mobx-react";
import {toJS} from "mobx";
import './App.scss';

class App extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      text: '',
      results: []
    }

  }

  sendText = async () => {
    await this.props.store.sendText(this.state.text);
    this.setState({text: ''});
  }

  handleChange = (event) => {
      this.setState({[event.target.name]: event.target.value});
  }
  
  render() {
    return (
      <div className="App">
          <Container fluid="lg">
            <Row>
              <Col xs={6}>
                
                  <Form.Group>
                    <Form.Label>Text</Form.Label>
                    <Form.Control type="text"  onChange={this.handleChange} value={this.state.text} name="text" placeholder="Text" />
                  </Form.Group>
                  <Button variant="primary" type="button" onClick={this.sendText}>
                    Submit
                  </Button>
                
              </Col>
              <Col xs={6}>
                <h2>Words</h2>
                <ul>
                {toJS(this.props.store.getText).map(r => 
                  <li>{r}</li>
                 )}
                </ul>
              </Col>
            </Row>
          </Container>
      </div>
    );
  }
}

export default inject('store')(observer(App));
