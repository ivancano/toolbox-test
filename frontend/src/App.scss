@import "~bootstrap/scss/bootstrap";

body {
    margin: 0;
    padding: 0;
    font-family: 'Open Sans', sans-serif;
    color: #000;
    font-weight: 300;
}

.container-lg {
    height: 100vh;
    padding-top: 20px;
}
