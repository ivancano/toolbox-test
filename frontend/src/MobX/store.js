import {observable, action, computed, decorate} from 'mobx';
import Service from '../services/services.js';


class Store {

	words = [];

	get getText() {
        return this.words;
    }

    setText(text) {
    	this.words.push(text);
    }

	async sendText(text) {
		const res = await Service.sendText(text);
		if(!res.error){
			this.words.push(text);
		}
	}

}

decorate(Store, {
	words: observable,
	getText: computed,
    sendText: action,
    setText: action
});

const store = new Store();
export default store;
export {
	Store
}