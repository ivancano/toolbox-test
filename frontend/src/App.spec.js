import React from 'react';
import renderer, {act} from 'react-test-renderer';
import App from './App';
import store from './MobX/store';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App store={store}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
 
  it('renders the inner Counter', () => {
    const wrapper = mount(<App store={store}/>);
    wrapper.find('input').simulate('change', {target: {value: 'My new value'}});
    wrapper.find('Button').simulate('click');
  });
});