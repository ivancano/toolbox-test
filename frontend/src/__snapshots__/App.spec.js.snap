// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="App"
>
  <div
    className="container-lg"
  >
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
          >
            Text
          </label>
          <input
            className="form-control"
            name="text"
            onChange={[Function]}
            placeholder="Text"
            type="text"
            value=""
          />
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Submit
        </button>
      </div>
      <div
        className="col-6"
      >
        <h2>
          Words
        </h2>
        <ul />
      </div>
    </div>
  </div>
</div>
`;
